main.tf

```diff
@@ -194,3 +205,11 @@ locals {
     }
   } : null
 }
+
+data "ibm_database_connection" "database_connection" {
+  count         = length(var.users) > 0 ? length(var.users) : 0
```

Comment:
No need for this count. Every instance will have a hostname and a port regardless of whether creating credentials or not. but do we really need the data lookup? Is there no way to get from `ibm_database` resource ?

Reply:
Sure, I will remove the count. 

No, `connection_strings` attribute is deprecated now, and it's advised to use `ibm_database_connection`. See https://registry.terraform.io/providers/IBM-Cloud/ibm/latest/docs/resources/database#connectionstrings

Reply:
Wait, I cannot remove the count because I need `user_id` and `user_type` which can only be generated when `users` list is present.

Reply:
Then count needs to be updated to take into consideration the `service_credential_names` variables since these will also have hostname and port outputs. 
which reminds me, you still need to update the `service_credentials_object` output with `port`. See -> https://github.com/terraform-ibm-modules/terraform-ibm-icd-mongodb/blob/main/main.tf#L173

Reply:
what would be the dependency of service_credential_names for hostname and port outputs? Right now the connection string is based on the user id, user type for each db deployment and endpoint type

Reply:
The count logic is still wrong. Its going to do a count for every user. you only need it once: `length(var.users) > 0 ? 1 : 0`

Reply:
oh yes, actually I updated below the code to consume just the first value. But yes we can update the logic here too